docs/pages/tutorial/deploy-to-mainnet/main.md

```diff
@@ -0,0 +1,187 @@
+
+```toml
+[profile.sepolia]
+```
+
+* Create the [`dojo_sepolia.toml`](https://github.com/rsodre/512karat/blob/main/dojo/dojo_sepolia.toml) dojo config file, with the same contents of [`dojo_dev.toml`](https://github.com/rsodre/512karat/blob/main/dojo/dojo_dev.toml), except for `[env]`, in which we're going to expose the `world_address` only:
+
+```toml
+[env]
+# rpc_url = ""         # env: STARKNET_RPC_URL
+# account_address = "" # env: DOJO_ACCOUNT_ADDRESS
+# private_key = ""     # env: DOJO_PRIVATE_KEY
+# world_address = "<World Address>"
+```
+
+* It's recommended to keep the `world_address` empty, on the first deployment it will be outputed by the deployment script. Then you should expose it.
+
+* Clone the [`dev`](https://github.com/rsodre/512karat/blob/main/dojo/overlays/dev/) overlays to [`sepolia`](https://github.com/rsodre/512karat/blob/main/dojo/overlays/sepolia/)
+
+* Create `.env.sepolia` containing your RPC provider, account and private key. Make sure that account is deployed and has some [ETH](https://starknet-faucet.vercel.app) in it (0.001 is more than enough).
+
+```sh
+# usage: source .env.sepolia
+export STARKNET_RPC_URL=<RPC_PROVIDER_URL>
+export DOJO_ACCOUNT_ADDRESS=<YOUR_ACCOUNT_ADDRESS>
+export DOJO_PRIVATE_KEY=<YOUR_ACCOUNT_PRIVATE_KEY>
+```
+
+
+
+### Deployment
+
+* To load the env variables and deploy to the chain you can use this script (for mainnet, just replace `sepolia` with `mainnet`):
+
+```bash
+# Stop script on error
+set -e
+
+# Load environment variables from the appropriate file
+ENV_FILE=".env.sepolia"
+
+if [ -f "$ENV_FILE" ]; then
+  echo "Loading environment variables from $ENV_FILE..."
+  export $(grep -v '^#' "$ENV_FILE" | xargs)
+else
+  echo "Environment file $ENV_FILE not found!"
+  exit 1
+fi
+
+# Define a cleanup function to clear environment variables
+cleanup_env() {
+  echo "Cleaning up environment variables..."
+  unset STARKNET_RPC_URL
+  unset DOJO_ACCOUNT_ADDRESS
+  unset DOJO_PRIVATE_KEY
+  echo "Environment variables cleared."
+}
+
+# Set the trap to execute cleanup on script exit or error
+trap cleanup_env EXIT
+
+# Build the project
+echo "Building the project..."
+sozo -P sepolia build
+
+# Deploy the project
+echo "Deploying to Sepolia..."
+sozo -P sepolia migrate
+
+# Deployment succeeded message
+echo "Deployment completed successfully."
+```
+
+* For this script to work don't forget to give it execution permissions:
+ `chmod +x <script_name>.sh`
+
+* This localises the env variables to the deployment script, so if for any reason the deployment is aborted, it cleans up the env variables.
+
+
+* sozo will output the rpc url, account address and deployed block.
+
+```sh
+ profile | chain_id | rpc_url
+---------+----------+------------------------
+ sepolia | SN_SEPOLIA | <RPC_PROVIDER_URL>
+
+
+🌍 World deployed at block <DEPLOYED_BLOCK> with txn hash: <DEPLOYMENT_TXN_HASH>
+⛩️  Migration successful with world at address <WORLD_ADDRESS>
+```
+
+Your world is deployed! 
```

Comment:
Once the world is deployed, currently we need to add the `world_block` in the `dojo_<PROFILE>.toml` file. May be worth mentioning it here.

Reply:
This will no longer be necessary in the future, but it is for now on public networks with a lot of blocks (mainnet).

Reply:
How do you add it to the toml? Didn't even know that lol. is it just:
```toml
[env]
rpc_url = "https://api.cartridge.gg/x/starknet/mainnet"  
# account_address = "" # env: DOJO_ACCOUNT_ADDRESS
# private_key = ""     # env: DOJO_PRIVATE_KEY
world_address = "<world address>"
starting_block = 129321
```